### PR TITLE
allow accessing Studio viewer mode using groupId, artifactId, and versionId (GAV coordinates)

### LINKED ISSUE #617: Feature Request: `DataSpace` Enterprise view (MVP)
### Overview

Enterprise view is a curated area to show dataspaces organized in a taxonomy tree to let users navigate concepts. We will make use of the `meta::pure::profiles::enterprise profile` tag, introduced in https://github.com/finos/legend-pure/pull/354

The general idea is to tag data spaces with `meta::pure::profiles::enterprise profile.taxonomyNodes` where the value can be comma-separated list of `taxonomy` path. In `enterprise` model explorer view, we will index all found data spaces with the mentioned tagged value to build a taxonomy tree

### Implementation plan

- [x] Add placeholder tabs for DataSpace viewer - show `Work in Progress` in the content - Done in https://github.com/finos/legend-studio/pull/639
  - [x] Entitlement/Authorization
  - [x] Test Data
  - [x] A tab to navigate to the project of the dataspace + the project the data space targets - this relies on https://github.com/finos/legend-studio/issues/638
- [x] Add enterprise view for dataspace
  - [x] Fetch dataspaces and organize them, extract taxonomy information from tagged value `meta::pure::profiles::enterprise profile.taxonomyNodes`
  - [x] Build and render taxonomy tree
  - [x] Genericize `DataSpaceViewer` and `DataSpaceViewerState` to be reused in enterprise model explorer. 
  - [x] Panel view for taxonomy node's dataspaces
  - [x] Show the dataspace counts for each node
  - [x] Navigate to taxonomy node using URL path params
  - [x] Navigate to taxonomy node and dataspace using URL path params

### Additional context

For a better context on `DSL DataSpace`, see #469

### LINKED ISSUE #638: Feature request: Allow going to viewer mode using `groupId`, `artifactId`, and `versionId`
### Similar requests

- [X] I have searched and found no existing similar requests

### What problems are you trying to solve?

We should have a way to go to Studio viewer mode using `GAV` coordinate, i.e. `groupId`, `artifactId`, `versionId`

### Describe the solution you would like to see implemented

- [x] Allow another URL pattern to go to this viewer mode: `/-/view/gav/${groupId}:${artifactId}:${versionId}`
- [x] Allow specifying `latest` (for latest version) and `HEAD` for latest revision as `versionId`
- [x] Allow opening specific entity by adding `/entity/${entityPath}` to agreed URL
- [x] Since this flow will be significantly different than the other flows where we can build dependencies from `SDLC` `project configuration`, we would need to adjust `Viewer` graph building algorithm slightly. We currently use `SDLC` when fetching dependencies and use `depot` to fetch all upstream dependency entities. For this flow, we can just use `Depot` for both. 
- [x] Create graph building logic for this new flow (see what we do in `QueryStore`)
- [x] Hide project information in status bar and header of explorer side-bar
- [x] Disable viewing the project config in this mode

### Query App
- [x] Add `studio.url` to config
- [x] Provide link to open query project in `Studio`

### Describe alternatives you have considered

_No response_

### Documentation, Design, Adoption, Migration Strategy

_No response_

### Contribution

- [x] I would like to work on this feature

### PR DESCRIPTION
**Pull request by akphi**
## Summary

- [x] Completes some items in #617
- [x] Closes #638 
- [x] Minor improvement for `Feature Request` template

## How did you test this change?

https://user-images.githubusercontent.com/13574879/139949454-111ac7cb-73dc-4b05-9ba6-92890696177b.mov

### CODE REVIEW

**Comment by akphi** [comment]
@MauricioUyaguari brought up a good point, we will try to see if there's a way to configure the route in Studio to not require `sdlcServerKey`. Will investigate that post this PR

File: .changeset/brave-cows-sort.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-extension-dsl-data-space': patch
+---
+
+Allow viewing more informations about a data space when creating a new query in `Legend Query`, such as entitlement, project links (the project that the data space resides as well as the project that the data space refers to), test data, etc.
```
Comment (MauricioUyaguari) [inline .changeset/brave-cows-sort.md:5]: What entitlement work did you do here. Don't see any within the dsl data space.

Comment (akphi) [reply .changeset/brave-cows-sort.md:5]: The tab using `KeyIcon`, it's a placeholder, so meant to be WIP

File: packages/legend-query/src/stores/QueryStore.ts [hunk 2]
```diff
@@ -73,13 +74,19 @@ import type { QueryConfig } from '../application/QueryConfig';
 export const LATEST_VERSION_ALIAS = 'latest';
 export const LATEST_SNAPSHOT_VERSION_ALIAS = 'HEAD';
 
+interface QueryProjectInfo {
+  groupId: string;
+  artifactId: string;
+  versionId: string;
+}
 export abstract class QueryInfoState {
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-query/src/stores/QueryStore.ts:82]: wow.

File: packages/legend-query/src/stores/QueryStore.ts [hunk 3]
```diff
@@ -73,13 +74,19 @@ import type { QueryConfig } from '../application/QueryConfig';
 export const LATEST_VERSION_ALIAS = 'latest';
 export const LATEST_SNAPSHOT_VERSION_ALIAS = 'HEAD';
 
+interface QueryProjectInfo {
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-query/src/stores/QueryStore.ts:77]: Potentially, except that I think the use case is not generic enough to justify putting it there, if this pattern of usage comes up more, we would put this in `DepotUtils` or something like that. It should stay here for now.

File: packages/legend-server-depot/src/DepotUtils.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+export const parseGAVCoordinates = (
+  gav: string,
+): {
+  groupId: string;
+  artifactId: string;
+  versionId: string;
+} => {
+  const parts = gav.split(GAV_DELIMITER);
+  assertTrue(parts.length === 3, `Can't parse GAV coordinate '${gav}'`);
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-server-depot/src/DepotUtils.ts:40]: will do!

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**